gitwash/development_workflow.rst

```diff
@@ -273,35 +220,138 @@ your history will look like this:
                  /
     D---E---F---G trunk
 
-See `rebase without tears`_ for more detail.
 
-To do a rebase on trunk::
+In git, rebasing is a mild form of re-writing history: it effectively
+forwards all your commits to the updated upstream commit. For a much
+more detailed explanation (with pictures!) see `this nice write up
+<https://git-scm.com/book/en/Git-Branching-Rebasing>` or See `rebase
+without tears`_.  In general, re-writing history, particularly
+published history, is considered bad practice, but in this case it is
+very useful.
 
-    # Update the mirror of trunk
-    git fetch upstream
-    # go to the feature branch
-    git checkout cool-feature
-    # make a backup in case you mess up
-    git branch tmp cool-feature
-    # rebase cool-feature onto trunk
-    git rebase --onto upstream/master upstream/master cool-feature
+The following example assumes that the remote of _your_ GitHub
+repository is called ``origin`` and the remote of the official
+repository is called ``upstream``.
 
-In this situation, where you are already on branch ``cool-feature``, the last
-command can be written more succinctly as::
+The first step is to make sure that your local copy of the upstream repository is
+up-to-date::
 
-    git rebase upstream/master
+     git fetch upstream
 
-When all looks good you can delete your backup branch::
+This updates your local copy of the repository, but does not change any files
+in your working copy.  Next, switch to the branch that you want to update and make
+a backup (just to be safe!).::
 
-   git branch -D tmp
+     git checkout cool-feature
+     git branch tmp cool-feature
+
+You are now ready to start the rebase of your branch onto the target
+parent branch, in this case `matplotlib/master` ::
+
+     git rebase upstream/master
+
+If there are no conflicts git will print something like::
```

Comment:
I think this would benefit from 

```
 .. code-block:: none
```

before, to prevent default bash markup.  Likewise for the non-bash stuff below.